src/analysis/convex/body.lean

```diff
@@ -0,0 +1,154 @@
+  [has_continuous_const_smul α β] (a : α) {s : set β}
+  (hs : is_compact s) : is_compact (a • s) := hs.image (continuous_id'.const_smul a)
+
+-- MOVE
+instance nnreal.has_continuous_const_smul
+  {α : Type} [mul_action ℝ α] [topological_space α]
+  [has_continuous_const_smul ℝ α] : has_continuous_const_smul ℝ≥0 α :=
+{ continuous_const_smul := λ c,
+  begin
+    convert continuous_id'.const_smul (↑c : ℝ),
+    apply_instance
+  end }
+
+-- needed to help typeclass resolution...
+-- localize this?
+--instance normed_space.has_continuous_const_smul_real : has_continuous_const_smul ℝ V :=
+--infer_instance
+
+namespace convex_body
+
+variables {V}
+
+instance : set_like (convex_body V) V :=
+{ coe := convex_body.carrier,
+  coe_injective' := λ K L h, by { cases K, cases L, congr' } }
+
+lemma convex (K : convex_body V) : convex ℝ (K : set V) := K.convex'
+lemma is_compact (K : convex_body V) : is_compact (K : set V) := K.is_compact'
+lemma nonempty (K : convex_body V) : (K : set V).nonempty := K.nonempty'
+
+@[ext]
+protected lemma ext {K L : convex_body V} (h : (K : set V) = L) : K = L := set_like.ext' h
+
+@[simp]
+lemma coe_mk (s : set V) (h₁ h₂ h₃) : (mk s h₁ h₂ h₃ : set V) = s := rfl
+
+instance : add_monoid (convex_body V) :=
+{ add := λ K L, ⟨(K : set V) + L,
+                 K.convex.add L.convex,
+                 K.is_compact.add L.is_compact,
+                 K.nonempty.add L.nonempty⟩,
+  add_assoc := λ K L M, by { ext, simp only [coe_mk, add_assoc] },
+  zero := ⟨0, convex_singleton 0, is_compact_singleton, set.singleton_nonempty 0⟩,
+  zero_add := λ K, by { ext, simp only [coe_mk, zero_add] },
+  add_zero := λ K, by { ext, simp only [coe_mk, add_zero] } }
+
+@[simp]
+lemma coe_add (K L : convex_body V) : (↑(K + L) : set V) = (K : set V) + L := rfl
+
+@[simp]
+lemma coe_zero : (↑(0 : convex_body V) : set V) = 0 := rfl
+
+instance : inhabited (convex_body V) := ⟨0⟩
+
+instance : add_comm_monoid (convex_body V) :=
+{ add_comm := λ K L, by { ext, simp only [coe_add, add_comm] },
+  .. convex_body.add_monoid }
+
+instance : has_smul ℝ≥0 (convex_body V) :=
+{ smul := λ c K, ⟨c • (K : set V), K.convex.smul _, K.is_compact.smul _, K.nonempty.smul_set⟩}
+
+@[simp]
+lemma coe_smul (c : ℝ≥0) (K : convex_body V) : (↑(c • K) : set V) = c • (K : set V) := rfl
+
+instance : mul_action ℝ≥0 (convex_body V) :=
+{ to_has_smul := convex_body.has_smul,
+  one_smul := λ K, by { ext, simp only [coe_smul, one_smul] },
+  mul_smul := λ c d K, by { ext, simp only [coe_smul, mul_smul] } }
+
+instance : distrib_mul_action ℝ≥0 (convex_body V) :=
+{ to_mul_action := convex_body.mul_action,
+  smul_add := λ c K L, by { ext, simp only [coe_smul, coe_add, smul_add] },
+  smul_zero := λ c, by { ext, simp only [coe_smul, coe_zero, smul_zero] } }
+
+/--
+The convex bodies in a fixed space $V$ form a module over the nonnegative reals.
+-/
+instance : module ℝ≥0 (convex_body V) :=
+{ to_distrib_mul_action := convex_body.distrib_mul_action,
+  add_smul := λ c d K,
```

Comment:
You could just use [convex.add_smul](https://leanprover-community.github.io/mathlib_docs/analysis/convex/basic.html#convex.add_smul), right?

Reply:
Ahh. Thanks for the tip!